.github/workflows/wheels.yml

```diff
@@ -0,0 +1,165 @@
+# Workflow to build and test wheels
+name: Wheel builder
+
+on:
+  schedule:
+    # Nightly build at 3:42 A.M.
+    - cron: "42 3 */1 * *"
+  push:
+    tags:
+      # Final releases, release candidates and post-releases build
+      - "[0-9]+.[0-9]+.[0-9]+(rc[0-9]+)?(.post[0-9]+)?"
+  pull_request:
+    branches:
+      - master
+      - "[0-9]+.[0-9]+.X"
+
+jobs:
+  # Check whether to trigger the build
+  check_build:
+    name: Check build trigger
+    runs-on: ubuntu-latest
+    outputs:
+      build: ${{ steps.check_build.outputs.build }}
+
+    steps:
+      - name: Checkout scikit-learn
+        uses: actions/checkout@v1
+
+      - name: Set environment variables
+        run: |
+          echo "COMMIT_MSG=$(git log --no-merges -1 --oneline)" >> $GITHUB_ENV
+
+      - id: check_build
+        name: Check build
+        if: endsWith(env.COMMIT_MSG, '[cd build]') || github.event_name == 'push'
+        run: |
+          echo "::set-output name=build::true"
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }} for Python ${{ matrix.python }}
+    runs-on: ${{ matrix.os }}
+    needs: check_build
+    if: needs.check_build.outputs.build
+
+    strategy:
+      # Avoid that the wheel builder for a
+      # runner is cancelled if other fails
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        # Build the wheels for Python 3.6 and newer
+        python: [36, 37, 38]
+
+    steps:
+      - name: Checkout scikit-learn
+        uses: actions/checkout@v1
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+
+      - name: Install dependencies
+        shell: bash
+        # OpenMP is not present on macOS by default
+        run: |
+          if [ "$RUNNER_OS" == "macOS" ]; then
+            brew install libomp
+            echo "CC=/usr/bin/clang" >> $GITHUB_ENV
+            echo "CXX=/usr/bin/clang++" >> $GITHUB_ENV
+            echo "CPPFLAGS=$CPPFLAGS -Xpreprocessor -fopenmp" >> $GITHUB_ENV
+            echo "CFLAGS=$CFLAGS -I/usr/local/opt/libomp/include" >> $GITHUB_ENV
+            echo "CXXFLAGS=$CXXFLAGS -I/usr/local/opt/libomp/include" >> $GITHUB_ENV
+            echo "LDFLAGS=$LDFLAGS -Wl,-rpath,/usr/local/opt/libomp/lib -L/usr/local/opt/libomp/lib -lomp" >> $GITHUB_ENV
+          fi
+          python -m pip install cibuildwheel
+
+      - name: Build and test wheels
+        env:
+          CIBW_BUILD: cp${{ matrix.python }}-*
+          CIBW_TEST_REQUIRES: pytest pandas
+          CIBW_TEST_COMMAND: pytest --pyargs sklearn
+          # Embed vcomp140.dll before generating the Windows
+          # wheel and after building the package from source
+          CIBW_BEFORE_BUILD_WINDOWS: pip install numpy scipy cython &&
+                                     rm -rf build &&
+                                     python setup.py build &&
+                                     python build_tools/github/vendor_vcomp140.py
```

Comment:
The the build isolation takes this into account when we run `pip wheel`?

Reply:
As far as I can tell yes. We need to use `pip wheel`. We probably shouldn't run `python setup.py build` but than it's true that vendoring `vcomp140` becomes harder. Couldn't it be done as part of the CI via some env variables?

Reply:
IIUC https://github.com/scikit-learn/scikit-learn/pull/17921#issuecomment-720703482, the idea is to keep the `vendor_vcomp140.py` script and then check that the generated wheel has the `vcomp140` dll?